engine/src/main/scala/cromwell/engine/workflow/lifecycle/deletion/DeleteWorkflowFilesActor.scala

```diff
@@ -223,10 +224,20 @@ class DeleteWorkflowFilesActor(rootWorkflowId: RootWorkflowId,
     womValue.collectAsSeq({ case womSingleFile: WomSingleFile => womSingleFile })
   }
 
+  /**
+    * Returns Paths for WomSingleFiles in allOutputs that are not in finalOutputs, verifying that the Paths are contained
+    * in a workflow execution directory associated with the root workflow (as a sanity check to avoid deleting files
+    * outside of the workflow execution directory).
+    */
   def gatherIntermediateOutputFiles(allOutputs: Set[WomValue], finalOutputs: Set[WomValue]): Set[Path] = {
     val allOutputFiles = allOutputs.flatMap(getWomSingleFiles)
     val finalOutputFiles = finalOutputs.flatMap(getWomSingleFiles)
-    allOutputFiles.diff(finalOutputFiles).flatMap(toPath)
+    val potentialIntermediaries = allOutputFiles.diff(finalOutputFiles).flatMap(toPath)
+    val checkedIntermediaries = potentialIntermediaries.filter(p => rootWorkflowRoots.exists(r => p.pathAsString.startsWith(r.pathAsString)))
```

Comment:
Is this "startsWith" check robust enough? Should I do be doing any sort of absolute path resolution?

Reply:
I think in general the best way is to compare canonicalized paths (I don't have an example handy but the codebase should yield some)

Reply:
I only see "canonical" in reference to Java Files (where I agree it's definitely the right method to use). Since these are our own Path objects, and they can refer to things like gcs paths, are we able to use the Java File getCanonicalPath method? 

Our Path class does have a toAbsolutePath method, so I switched to using that and then the Path class startsWith method. I think that should be good, because there are Path-specific implementations of both methods (CloudStoragePath in this case). In reality though, I would expect that we are already dealing with absolute paths.